### PR TITLE
chore(go-sdk): move utils to internalUtils

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Avoid name collision between file and directory

## References
Close https://github.com/openfga/go-sdk/issues/30
Corresponding GOSDK changes: https://github.com/openfga/go-sdk/pull/31

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Review by rhamzeh — COMMENTED** [review]
Can you call this folder/package `internalutils` instead? `internal-utils` also works for the folder name

**Review by rhamzeh — COMMENTED** [review]
If it's OK with you, I think what @jpadilla was suggesting naming them internal/utils/ulid.go etc..

No necessary to do it in the scope of this PR, but just as an FYI

**Review by rhamzeh — DISMISSED** [review]

File: config/clients/go/config.overrides.json [hunk 1]
```diff
@@ -64,12 +64,12 @@
...
     "oauth2/token_test.go": {},
-    "utils/utils.mustache": {
-      "destinationFilename": "utils/utils.go",
+    "internal/utils/utils.mustache": {
+      "destinationFilename": "internal/utils/utils.go",
       "templateType": "SupportingFiles"
     },
-    "utils/utils_test.mustache": {
-      "destinationFilename": "utils/utils_test.go",
+    "internal/utils/utils_test.mustache": {
```
Comment (rhamzeh) [inline config/clients/go/config.overrides.json:71]: This doesn't seem to match the generated SDK 🤔

Comment (adriantam) [reply config/clients/go/config.overrides.json]: Did not push the changes :P

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**